src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory_Methods.cs

```diff
@@ -53,12 +53,33 @@ internal IArgument CreateArgumentOperation(ArgumentKind kind, IParameterSymbol p
 
         private IVariableDeclaration CreateVariableDeclarationInternal(BoundLocalDeclaration boundLocalDeclaration, SyntaxNode syntax)
         {
-            return OperationFactory.CreateVariableDeclaration(boundLocalDeclaration.LocalSymbol, Create(boundLocalDeclaration.InitializerOpt), _semanticModel, syntax);
+            IVariableInitializer initializer = null;
+            if (boundLocalDeclaration.InitializerOpt != null)
+            {
+                IOperation initializerValue = Create(boundLocalDeclaration.InitializerOpt);
+                SyntaxNode initializerSyntax;
+                bool isImplicit;
+                if (syntax is VariableDeclaratorSyntax variableDeclarator)
+                {
+                    initializerSyntax = variableDeclarator.Initializer;
```

Comment:
>variableDeclarator.Initializer [](start = 40, length = 30)

I think it would be good to ensure that this value is not null.

Reply:
I filed a bug to track investigating if the syntax can be something else for error cases. If not, we can remove this if check and convert it to a direct cast.

Reply:
https://github.com/dotnet/roslyn/issues/22574

Reply:
I think it would be better for this code to be defensive.

Reply:
Do you propose adding a Debug.Fail in the else part?

Reply:
> Do you propose adding a Debug.Fail in the else part?

That would be fine, asserts are good.

Reply:
It looks like we are still not protecting against a situation when ```initializerSyntax``` is null at the end.